src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -282,6 +375,182 @@ private boolean validatePasswords(String identityStorePass,
+            novelProvidersExist = true;
+            String[] novelProviders = populateArray(novelProvidersString);
+            addToProvidersMap(novelProviders, PROP_SECRET_PROVIDERS);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Identified the providers listed under secretProviders property of the secret-conf.");
+            }
+        }
+    }
+
+    /**
+     * Terminates if either properties, secretRepositories or secretProviders haven`t been configured.
+     */
+    private boolean isSecureVaultStatusValidated() {
+
+        if (!(legacyProvidersExist || novelProvidersExist)) {
+            if (log.isDebugEnabled()) {
+                log.debug("No secret provider has been configured.");
+            }
+            return false;
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Validated the secure vault status by identifying the configured secret providers.");
+        }
+        return true;
+    }
+
+    /**
+     * Util method to add all the providers from providers array to providers hash map along with the
+     * type (secretRepositories or secretProviders).
+     *
+     * @param providersArray Repositories array and secretProviders array generated from the properties,
+     *                       secretRepositories or secretProviders.
+     * @param providerType   SecretRepositories or secretProviders.
+     */
+    private void addToProvidersMap(String[] providersArray, String providerType) {
+
+        for (String providerName : providersArray) {
+            if (log.isDebugEnabled()) {
+                log.debug("Added a secret provider to providers map: " + providers);
+            }
+            providers.put(providerName, providerType);
+        }
+    }
+
+    /**
+     * Util method to add the split string of properties, secretRepositories or secretProviders to the Array.
+     *
+     * @param propertyValue Value of the property in the secret configuration file.
+     * @return An array containing the string.
+     */
+    private String[] populateArray(String propertyValue) {
+
+        String[] propertyValues = propertyValue.split(",");
+        if (propertyValues.length == 0) {
+            if (log.isDebugEnabled()) {
+                log.debug("No secret repositories have been configured.");
+            }
+        }
+        return propertyValues;
+    }
+
+    /**
+     * Creates the TrustKeyStoreWrapper and the IdentityKeyStoreWrapper.
+     *  @param identityKeyStoreWrapper Represents the private keyStore entry.
+     * @param trustKeyStoreWrapper    Represents the abstraction for trusted KeyStore.
+     * @param properties              Configuration properties.
+     * @return true upon successful creation of KeyStoreWrappers.
+     */
+    private boolean createKeyStoreWrappers(IdentityKeyStoreWrapper identityKeyStoreWrapper,
+                                           TrustKeyStoreWrapper trustKeyStoreWrapper, Properties properties) {
+
```

Comment:
499 -531 brought the logic resides inside the init method to a new method createKeyStoreWrappers().